src/main/java/gregtech/common/mui/widget/orefilter/OreFilterTestSlot.java

```diff
@@ -1,95 +1,85 @@
-package gregtech.common.gui.widget.orefilter;
-
-import gregtech.api.gui.GuiTextures;
-import gregtech.api.gui.IRenderContext;
-import gregtech.api.gui.resources.TextureArea;
-import gregtech.api.gui.widgets.ImageWidget;
-import gregtech.api.gui.widgets.WidgetGroup;
-import gregtech.api.util.LocalizationUtils;
-import gregtech.api.util.Position;
+package gregtech.common.mui.widget.orefilter;
+
+import gregtech.api.unification.OreDictUnifier;
 import gregtech.api.util.function.BooleanConsumer;
 import gregtech.api.util.oreglob.OreGlob;
+import gregtech.common.covers.filter.oreglob.impl.ImpossibleOreGlob;
 
-import net.minecraft.client.renderer.GlStateManager;
-import net.minecraft.client.resources.I18n;
-import net.minecraft.item.ItemStack;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
-
+import com.cleanroommc.modularui.api.drawable.IKey;
+import com.cleanroommc.modularui.widgets.ItemSlot;
 import it.unimi.dsi.fastutil.objects.Object2BooleanAVLTreeMap;
 import it.unimi.dsi.fastutil.objects.Object2BooleanMap;
 import it.unimi.dsi.fastutil.objects.Object2BooleanMaps;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.Set;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /**
  * @author brachy84
  */
-public abstract class OreFilterTestSlot extends WidgetGroup {
-
-    private final ImageWidget match;
-    private final ImageWidget noMatch;
+public class OreFilterTestSlot extends ItemSlot {
```

Comment:
This needs to set the max slot limit I believe, as I was able to insert a stack of items into the testing slot in the filter, and then was not able to clear the slot with left click. Plus it would just look better always being 1 item in the slot.